ubtue_University of Chicago Press Journal.js

```diff
@@ -0,0 +1,292 @@
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2020-08-21 17:30:16"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+	Copyright © 2020 Universitätsbibliothek Tübingen.  All rights reserved.
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+	You should have received a copy of the GNU Affero General Public License
+	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.includes('/doi/')) return "journalArticle";
+		else if (url.includes('/toc/') && getSearchResults(doc)) return "multiple";
+	else return false;
+}
+
+function getSearchResults(doc) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//*[contains(concat( " ", @class, " " ), concat( " ", "issue-item__title", " " ))]//a');
+	for (let row of rows) {
+		let href = row.href;
+		let title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+/*function invokeEMTranslator(doc) {
+	var translator = Zotero.loadTranslator("web");
+	translator.setTranslator("951c027d-74ac-47d4-a107-9c3069ab7b48");
+	translator.setDocument(doc);
+	translator.setHandler("itemDone", function (t, i) {
+		//scrape abstract from website instead of EM
+		let abstractFull = ZU.xpathText(doc, '//*[contains(concat( " ", @class, " " ), concat( " ", "abstractInFull", " " ))]//p');
+		if (abstractFull) i.abstractNote = abstractFull;
+		i.complete();
+	});
+	translator.translate();
+}*/
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) === "multiple") {
+		Zotero.selectItems(getSearchResults(doc), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else
+		scrape(doc, url);
+}
+
+function scrape(doc, url) {
+	var risURL = "https://www.journals.uchicago.edu/action/downloadCitation";
+	var doi = ZU.xpathText(doc, '//meta[@name="dc.Identifier" and @scheme="doi"]/@content');
+	if (!doi) {
+		doi = url.match(/10\.[^?#]+/)[0];
+	}
+	var post = "doi=" + encodeURIComponent(doi) + "&include=abs&format=ris&direct=false&submit=Download+Citation";
+	//Z.debug(risURL)
+
+	ZU.doPost(risURL, post, function (text) {
+		var translator = Zotero.loadTranslator("import");
+		translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
+		translator.setString(text);
+		translator.setHandler("itemDone", function (obj, item) {
+			Z.debug(text)
```

Comment:
Debug line, should be removed (or at least line should end with semicolon)

Reply:
removed